core/backend/src/ViewStateHydrator.ts

```diff
@@ -32,12 +35,35 @@ export class ViewStateHydrater {
     return response;
   }
 
+  private async handleCategoryIds(response: HydrateViewStateResponseProps, categoryIds: CompressedId64Set) {
+    // consider splitting up categoryIds, as queries get slow with many many categoryids in them.
+    const maxCategoriesPerQuery = 200;
+    const decompressedIds = CompressedId64Set.decompressArray(categoryIds);
+    const result = [];
+    while (decompressedIds.length !== 0) {
+      const end = (decompressedIds.length > maxCategoriesPerQuery) ? maxCategoriesPerQuery : decompressedIds.length;
+      const where = decompressedIds.splice(0, end).join(",");
+      const query = `SELECT ECInstanceId as id, Parent.Id as parentId, Properties as appearance FROM BisCore.SubCategory WHERE Parent.Id IN (${where})`;
+      try {
+        for await (const row of this._imodel.query(query, undefined, { rowFormat: QueryRowFormat.UseJsPropertyNames })) {
+          result.push(row as SubCategoryResultRow);
+        }
+      } catch {
+        // ###TODO: detect cases in which retry is warranted
```

Comment:
Copy-pasted comment not meaningful in backend.

Reply:
changed it